### PR TITLE
[diffusion] model: support a new model

### PR DESCRIPTION
**Pull request by nv-dmajchrowski**
## Motivation

This PR adds initial support for for a new world model (not yet released) that serves text-to-video (T2V), image-to-video (I2V), and text-to-image (T2I) generation from a single checkpoint.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #26437131059](https://github.com/sgl-project/sglang/actions/runs/26437131059)
Latest PR Test (Extra): :x: [Run #26437130836](https://github.com/sgl-project/sglang/actions/runs/26437130836)

### CODE REVIEW

**Comment by nvpohanh** [comment]
@Fridge003 could you help us to assign this to the SGLang-diffusers PIC for review? Thanks!

File: python/sglang/multimodal_gen/runtime/models/dits/cosmos3video.py [hunk 1]
```diff
@@ -0,0 +1,1468 @@
...
+)
+from sglang.multimodal_gen.runtime.loader.utils import get_param_names_mapping
+from sglang.multimodal_gen.runtime.models.dits.base import CachableDiT
+from sglang.multimodal_gen.runtime.utils.logging_utils import init_logger
+from sglang.srt.utils import add_prefix
+
+logger = init_logger(__name__)
+
+
+def _get_sp_world_size_safe() -> int:
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/models/dits/cosmos3video.py]: could you move it to `parallel_stage.py`? what's the necessity of this

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3.py [hunk 2]
```diff
@@ -0,0 +1,880 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3.py:1]: move it to `model_specific_stages`?

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3_guardrails.py [hunk 3]
```diff
@@ -0,0 +1,461 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3_guardrails.py:1]: ditto

File: python/sglang/multimodal_gen/runtime/models/dits/cosmos3video.py [hunk 4]
```diff
@@ -0,0 +1,1468 @@
...
+        if token_noisy_mask is not None:
+            hidden_gen = hidden_gen + time_embed.unsqueeze(1) * token_noisy_mask
+        else:
+            hidden_gen = hidden_gen + time_embed.unsqueeze(1)
+
+        # Ensure cache dicts exist (for backwards compatibility)
+        self._ensure_cache_dicts()
+
+        # Use optimized SDPA backend (FlashAttention when available)
+        with torch.nn.attention.sdpa_kernel(self.sdpa_backends, set_priority=True):
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/models/dits/cosmos3video.py]: is it possible that it's compatible with other attention backends

File: python/sglang/multimodal_gen/runtime/models/dits/cosmos3video.py [hunk 5]
```diff
@@ -0,0 +1,1468 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/models/dits/cosmos3video.py:1]: would you adapt to `USPAttention`?

File: python/sglang/multimodal_gen/runtime/models/dits/cosmos3video.py (hunk 1 above)
Comment (nv-dmajchrowski) [reply python/sglang/multimodal_gen/runtime/models/dits/cosmos3video.py]: This one we can safely remove, it's artifact from first version of multi-gpu support.

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3.py (hunk 2 above)
Comment (nv-dmajchrowski) [reply python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3.py:1]: Moved

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3_guardrails.py (hunk 3 above)
Comment (nv-dmajchrowski) [reply python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3_guardrails.py:1]: Moved

File: python/sglang/multimodal_gen/runtime/models/dits/cosmos3video.py (hunk 5 above)
Comment (nv-dmajchrowski) [reply python/sglang/multimodal_gen/runtime/models/dits/cosmos3video.py:1]: Done, adapted the codebase to `USPAttention`

**Review by mickqian — APPROVED** [review]

File: python/sglang/multimodal_gen/runtime/models/dits/cosmos3video.py (hunk 4 above)
Comment (nv-dmajchrowski) [reply python/sglang/multimodal_gen/runtime/models/dits/cosmos3video.py]: I removed this wrapper, now we fallback to default `_supported_attention_backends` from `DiTArchConfig`

**Comment by nvpohanh** [comment]
@nv-dmajchrowski could you fix the conflicts so that I can trigger another pipeline? thanks!
